src/Annotation.cpp

```diff
@@ -0,0 +1,136 @@
+#include "Annotation.h"
+#include "Debug.h"
+#include "FunctionAnnotation.h"
+#include "ParameterAnnotation.h"
+#include "ReturnAnnotation.h"
+#include "StructAnnotation.h"
+#include "Utils.h"
+
+Annotation::Annotation() {
+  this->annotationType = AnnotationType::UndefinedAnnotation;
+}
+
+bool Annotation::annotationIsUndefined() {
+  return this->annotationType == AnnotationType::UndefinedAnnotation;
+}
+
+Annotation *generateAnnotation(const std::string &rawAnnotationString) {
```

Comment:
Is it possible that this method will ever be called with an invalid `rawAnnotationString` (e.g., one that the user wrote incorrectly)? I think it might be possible, in which case we need to do error checking and then emit a useful error message rather than just segfaulting (which I think would be the current behavior, for example if `chunks` ends up having 4 or fewer elements due to an invalid input).

I think the best fix for this is to write code that validates that a raw string meets the method's assumptions into a `checkValid` method that returns `bool`. Then, call that method immediately when you read one of these strings from the user, so that you can safely make these assumptions here. Then, you can add a comment at the top of this method that refers to that one, which will re-assure someone reading this code that the e.g., unguarded array access to `chunks[4]` at line 78 is safe, because the input string is guaranteed to be in the proper format.